verify-service/src/main/java/io/mosip/verifyservice/services/VerifiablePresentationSubmissionServiceImpl.java

```diff
@@ -0,0 +1,107 @@
+
+import static io.mosip.verifycore.utils.SecurityUtils.*;
+import static io.mosip.verifycore.utils.Utils.getVerificationStatus;
+
+@Service
+public class VerifiablePresentationSubmissionServiceImpl implements VerifiablePresentationSubmissionService {
+
+    @Autowired
+    AuthorizationRequestCreateResponseRepository authorizationRequestCreateResponseRepository;
+    @Autowired
+    VpSubmissionRepository vpSubmissionRepository;
+
+    @Override
+    public VpSubmissionResponseDto submit(VpSubmissionDto vpSubmissionDto) {
+        new Thread(() -> {
+            processSubmission(vpSubmissionDto);
+        }).start();
+        return new VpSubmissionResponseDto("", "", "");
+
+    }
+
+    private void processSubmission(VpSubmissionDto vpSubmissionDto) {
+        JSONObject vpProof = new JSONObject(vpSubmissionDto.getVpToken()).getJSONObject(Constants.KEY_PROOF);
+        String jws = getFormattedJws(vpProof.getString(Constants.KEY_JWS));
+        String publicKeyPem = vpProof.getString(Constants.KEY_VERIFICATION_METHOD);
+
+        //TODO: Dynamic algo type
+        try {
+            Algorithm algorithm = Algorithm.RSA256(getPublicKeyFromString(publicKeyPem), null);
+            JWTVerifier verifier = JWT.require(algorithm).build();
+            verifier.verify(jws);
+
+            JSONArray verifiableCredentials = new JSONObject(vpSubmissionDto.getVpToken()).getJSONArray(Constants.KEY_VERIFIABLE_CREDENTIAL);
+            List<VcResult> verificationResults = getVcVerificationResults(verifiableCredentials);
+            boolean combinedVerificationStatus = true;
+            for (VcResult verificationResult : verificationResults) {
+                combinedVerificationStatus = combinedVerificationStatus && (verificationResult.getVerificationStatus() == VerificationStatus.SUCCESS);
+            }
+            if (!combinedVerificationStatus) {
+                throw new VerificationFailedException();
+            }
+            vpSubmissionRepository.save(new VpSubmission(vpSubmissionDto.getState(), vpSubmissionDto.getVpToken(), vpSubmissionDto.getPresentationSubmission(), SubmissionStatus.SUCCESS));
+        } catch (Exception e) {
+            e.printStackTrace();
+            vpSubmissionRepository.save(new VpSubmission(vpSubmissionDto.getState(), vpSubmissionDto.getVpToken(), vpSubmissionDto.getPresentationSubmission(), SubmissionStatus.FAILED));
+        }
+
+        authorizationRequestCreateResponseRepository.findById(vpSubmissionDto.getState()).map(authorizationRequestCreateResponse -> {
+            authorizationRequestCreateResponse.setSubmissionState(SubmissionState.COMPLETED);
+            authorizationRequestCreateResponseRepository.save(authorizationRequestCreateResponse);
+            return null;
+        });
+    }
+
+    @Override
+    public VpTokenResultDto getSubmissionResult(String requestId, String transactionId) {
+        VpSubmission vpSubmissionResult = vpSubmissionRepository.findById(requestId).orElse(null);
```

Comment:
Are we not checking if vp_token is matching as the presentation definition ?

Reply:
For the demo we were not. This is a work in progress and will be part of the next PR.

Reply:
Jira task for the same, lets attach here

Reply:
https://mosip.atlassian.net/browse/INJIVER-846